bootstrap/bootstrap.go

```diff
@@ -431,11 +432,42 @@ func (b *Bootstrap) PluginPhase() error {
 		checkout, err := b.checkoutPlugin(p)
 		if err != nil {
 			return errors.Wrapf(err, "Failed to checkout plugin %s", p.Name())
-
+		}
+		if b.Config.PluginValidation && checkout.Definition == nil {
+			if b.Debug {
+				b.shell.Commentf("Parsing plugin definition for %q from %s", p.Label(), checkout.CheckoutDir)
+			}
+			// parse the plugin definition from the plugin checkout dir
+			checkout.Definition, err = plugin.LoadDefinitionFromDir(checkout.CheckoutDir)
+			if err == plugin.ErrDefinitionNotFound {
+				b.shell.Warningf("Failed to find plugin definition")
+			} else if err != nil {
+				return err
+			}
 		}
 		b.plugins = append(b.plugins, checkout)
 	}
 
+	if b.Config.PluginValidation {
+		for _, checkout := range b.plugins {
+			if b.Debug {
+				json, _ := json.Marshal(checkout.Definition)
```

Comment:
Will there be a definition if one couldn't be loaded? i.e. will `checkout.Definition` be nil and this code blow up?

Reply:
Yup, it will blow up. Good catch, added a check.